axiom/datasets.go

```diff
@@ -363,10 +363,14 @@ func (s *DatasetsService) Ingest(ctx context.Context, id string, r io.Reader, ty
 		return nil, spanError(span, err)
 	}
 
-	var res ingest.Status
-	if _, err = s.client.Do(req, &res); err != nil {
+	var (
+		res  ingest.Status
+		resp *Response
+	)
+	if resp, err = s.client.Do(req, &res); err != nil {
 		return nil, spanError(span, err)
 	}
+	res.TraceID = resp.TraceID()
```

Comment:
Do you think that there might be other things we'll want to extract from the Response down the line? Would it be too much to embed `*Response` in all these `Status` structs so that one can then just call `TraceID()`?

Reply:
`Response` itself embeds `*http.Response` so that would be exported to our users, as well. I don't like that idea too much, tbh as I prefer the very cleaned-up, self-explaining structs we export.

In the future we might want to clean that up but lets re-visit that in a follow-up PR.